src/utilities/fetchAndExtractZip.utils.ts

```diff
@@ -73,6 +73,31 @@ const fetchZipFromUrl = async (url: string): Promise<ArrayBuffer> => {
 	return zipData;
 };
 
+export const unpackFileZip = async (file: File | ArrayBuffer): Promise<ProcessedZipResult> => {
+	try {
+		const zip = new JSZip();
+		const content = await zip.loadAsync(file);
+		const structure = await processZipContent(content);
+
+		return { structure };
+	} catch (error) {
+		const errorMessage =
+			error instanceof Error
+				? error.message
+				: i18n.t("fetchAndExtract.uknownErrorUnpackingZip", { ns: "utilities" });
+		LoggerService.error(
+			namespaces.utilities.fetchAndExtract,
+			i18n.t("fetchAndExtract.fetchAndExtractError", {
```

Comment:
Missing file name and stack for the translation

Reply:
added

Reply:
nice!